supernode/node/grpc/server/services/walletnode/process_userdata.go

```diff
@@ -0,0 +1,338 @@
+	}
+	if req == nil {
+		return nil, errors.Errorf("Request receive from walletnode is nil")
+	}
+	// Convert protobuf request to UserdataProcessRequest
+	request := userdata.ProcessRequestSigned{
+		Userdata: &userdata.ProcessRequest{
+			RealName:        req.RealName,
+			FacebookLink:    req.FacebookLink,
+			TwitterLink:     req.TwitterLink,
+			NativeCurrency:  req.NativeCurrency,
+			Location:        req.Location,
+			PrimaryLanguage: req.PrimaryLanguage,
+			Categories:      req.Categories,
+			Biography:       req.Biography,
+			AvatarImage: userdata.UserImageUpload{
+				Content:  req.AvatarImage.Content,
+				Filename: req.AvatarImage.Filename,
+			},
+			CoverPhoto: userdata.UserImageUpload{
+				Content:  req.CoverPhoto.Content,
+				Filename: req.CoverPhoto.Filename,
+			},
+			ArtistPastelID:    req.ArtistPastelID,
+			Timestamp:         req.Timestamp,
+			PreviousBlockHash: req.PreviousBlockHash,
+		},
+		UserdataHash: req.UserdataHash,
+		Signature:    req.Signature,
+	}
+
+	processResult, err := task.SupernodeProcessUserdata(ctx, &request)
+	if err != nil {
+		return nil, errors.Errorf("SupernodeProcessUserdata can not process %w", err)
+	}
+	if processResult.ResponseCode == userdata.ErrorOnContent {
+		return &pbwn.UserdataReply{
+			ResponseCode:    processResult.ResponseCode,
+			Detail:          processResult.Detail,
+			RealName:        processResult.RealName,
+			FacebookLink:    processResult.FacebookLink,
+			TwitterLink:     processResult.TwitterLink,
+			NativeCurrency:  processResult.NativeCurrency,
+			Location:        processResult.Location,
+			PrimaryLanguage: processResult.PrimaryLanguage,
+			Categories:      processResult.Categories,
+			Biography:       processResult.Biography,
+			AvatarImage:     processResult.AvatarImage,
+			CoverPhoto:      processResult.CoverPhoto,
+		}, nil
+	}
+
+	if task.ConnectedTo != nil {
+		// This is secondary node, we just response to walletnode here
+		return &pbwn.UserdataReply{
+			ResponseCode: processResult.ResponseCode,
+			Detail:       processResult.Detail,
+		}, nil
+	}
+	// Process actual write to rqlite db happen here
+	<-task.NewAction(func(ctx context.Context) error {
+		if processResult.ResponseCode == userdata.SuccessVerifyAllSignature {
+			if service.databaseOps == nil {
+				processResult.ResponseCode = userdata.ErrorRQLiteDBNotFound
+				processResult.Detail = userdata.Description[userdata.ErrorRQLiteDBNotFound]
+				return errors.Errorf("databaseOps service object is empty")
+			}
+			// Send data to SN contain the leader rqlite
+			if !service.databaseOps.IsLeader() {
+				if task.ConnectedToLeader != nil {
+					if _, err := task.ConnectedToLeader.ProcessUserdata.SendUserdataToLeader(ctx, request); err != nil {
+						processResult.ResponseCode = userdata.ErrorWriteToRQLiteDBFail
+						processResult.Detail = userdata.Description[userdata.ErrorWriteToRQLiteDBFail]
+						return errors.Errorf("failed to send or write userdata to leader rqlite %w", err)
```

Comment:
Should always return nil in `task.NewAction` or it will kill other things. I know this is annoyed but we need to stick with it for now.

Reply:
solved, thank you